source/agora/consensus/protocol/Nominator.d

```diff
@@ -649,9 +649,15 @@ extern(D):
             Transaction[] received_tx_set;
             if (auto fail_reason = this.ledger.getValidTXSet(con_data, received_tx_set, this.ledger.getUTXOFinder()))
             {
-                log.info("Missing TXs while checking envelope signature : {}",
-                    scpPrettify(&envelope, &this.getQSet));
-                return; // We dont have all the TXs for this block. Try to catchup
+                if (fail_reason == this.ledger.InvalidConsensusDataReason.NotInPool)
+                {
+                    log.trace("Tx set being confirmed is not yet valid: {}", fail_reason);
+                    this.queued_envelopes.insertBack(envelope.clone()); // Put back in queue
+                    this.envelope_timer.rearm(nomination_interval, Periodic.No);
```

Comment:
`envelope_timer` is armed with `EnvTaskDelay` through out the code.

Reply:
We don't want to process this one until the `tx pool` contains all the missing `tx`. Missing `tx` are fetched from other nodes by the function `getUnknownTXs` that runs at an interval of `block_catchup_interval`.  I chose to use `nomination_interval` only because it is a longer duration than `EnvTaskDelay`.
However, I am now thinking that we need a smarter approach to not throw away this envelope but also only process it when we have all `txs`.

Reply:
I have changed the approach now to use `ValidationLevel.kMaybeValidValue` in `validateValue`.